compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -2768,7 +2768,8 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
       def SomeModule: Symbol = dotc.core.Symbols.defn.SomeClass.companionModule
       def ProductClass: Symbol = dotc.core.Symbols.defn.ProductClass
       def FunctionClass(arity: Int, isImplicit: Boolean = false, isErased: Boolean = false): Symbol =
-        dotc.core.Symbols.defn.FunctionSymbol(arity, isImplicit, isErased)
+        if isErased then dotc.core.Symbols.defn.ErasedFunctionClass
```

Comment:
Returning `ErasedFunction` is not really correct here. We have two options:
* Return `NoSymbol`
* Runtime failure if `isErased` is set

I would go for a runtime failure that tells the user that `ErasedFunctionN`/`ErasedFunctionFunctionN` do not exist anymore and explain the new encoding. To do this, we probably need to add an experimental `reflect.defn.ErasedFunctionClass` in `Quotes.scala`.